### ISSUE TITLE
Accessibility: Browse Group Categories block can have focusable hidden content

### ISSUE THREAD

**Issue by taylor-steve** [projectblacklight/spotlight#3578]
Running `axe-core-rspec` on `browse_group_categories_block_spec`:

```
     Failure/Error: expect(page).to be_axe_clean.within '#content'

       Found 1 accessibility violation:

       1) aria-hidden-focus: ARIA hidden element must not be focusable or contain focusable elements (serious)
           https://dequeuniversity.com/rules/axe/4.10/aria-hidden-focus?application=axeAPI
           The following 3 nodes violate this rule:

               Selector: #tns1-item3
               HTML: <div class="box category-1 justify-content-center justify-content-md-space-around tns-item" id="tns1-item3" aria-hidden="true" tabindex="-1">
               Fix all of the following:
               - Focusable content should have tabindex="-1" or be removed from the DOM

               Selector: #tns1-item4
               HTML: <div class="box category-1 justify-content-center justify-content-md-space-around tns-item" id="tns1-item4" aria-hidden="true" tabindex="-1">
               Fix all of the following:
               - Focusable content should have tabindex="-1" or be removed from the DOM

               Selector: #tns1-item5
               HTML: <div class="box category-1 justify-content-center justify-content-md-space-around tns-item" id="tns1-item5" aria-hidden="true" tabindex="-1">
               Fix all of the following:
               - Focusable content should have tabindex="-1" or be removed from the DOM

       Invocation: axe.run({"include"=>[["#content"]]}, {}, callback);
```
For a browse group category with more categories than will show:
<img width="1031" height="293" alt="Image" src="https://github.com/user-attachments/assets/9a61fa2b-65bc-4d20-8f86-a265f9728cb5" />

The hidden categories can be tabbed to:
<img width="1133" height="335" alt="Image" src="https://github.com/user-attachments/assets/7933325e-ad1f-406e-9ce8-373145968ab5" />
